cmd/thanos/query.go

```diff
@@ -431,6 +443,8 @@ func runQuery(
 			flagsMap,
 			instantDefaultMaxSourceResolution,
 			maxConcurrentQueries,
+			runtimeInfo,
+			buildInfo,
```

Comment:
Is there a reason to have this as an argument? Sounds like YAGNI. But, instead of moving this down to where it is actually needed, it's probably an even better idea to re-use this variable and pass it to `NewQueryUI()` as well, down to `q.status()` where `thanosVersion` is used. Then we would ensure that the same information is available in both UIs.

Reply:
Oh got it. So if we pass the `runtimeInfo` and `buildInfo` to `NewQueryUI()` then which `thanosVersion` struct should we remove. The one in /pkg/ui/query.go` or should I remove the one I added as you said before?

Reply:
I suppose let's remove the old one since it is now a part of the API, and then let's make those two variables in `cmd/thanos/query.go` and pass them down to where they are used to ensure that we don't do duplicate work

Reply:
Sure